plugins/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -0,0 +1,148 @@
+        self._hyperparameters = hyperparameters
+        self._train_parameters = task_config
+        self._label_column = label_column
+
+        outputs = {
+            "model": JoblibSerializedFile,
+            "predictions": List[float],
+            "evaluation_result": Dict[str, Dict[str, List[float]]],
+        }
+
+        super(XGBoostTask, self).__init__(
+            name,
+            task_type=self._TASK_TYPE,
+            task_config=task_config,
+            interface=Interface(inputs=inputs, outputs=outputs),
+            **kwargs,
+        )
+
+    # Train method
+    def train(self, dtrain, dvalid, **kwargs) -> Tuple[str, Dict[str, Dict[str, List[float]]]]:
+        evals_result = {}
+        # if validation data is provided, then populate evals and evals_result
+        booster_model = (
+            xgboost.train(
+                params=self._hyperparameters,
+                dtrain=dtrain,
+                **asdict(self._train_parameters if self._train_parameters else TrainParameters()),
+                evals=[(dvalid, "validation")],
+                evals_result=evals_result,
+            )
+            if dvalid
+            else xgboost.train(
+                params=self._hyperparameters,
+                dtrain=dtrain,
+                **asdict(self._train_parameters if self._train_parameters else TrainParameters()),
+            )
+        )
+        fname = "booster_model.joblib.dat"
+        joblib.dump(booster_model, fname)
+        return fname, evals_result
+
+    # Test method
+    def test(self, booster_model, dtest, **kwargs) -> List[float]:
+        booster_model = joblib.load(booster_model)
+        y_pred = booster_model.predict(dtest).tolist()
+        return y_pred
+
+    def execute(self, **kwargs) -> Any:
+        train_key = ""
+        test_key = ""
+        validation_key = ""
+
+        # fetch inputs
+        for key in self.python_interface.inputs.keys():
+            if "train" in key:
+                train_key = key
+            elif "test" in key:
+                test_key = key
+            elif "valid" in key:
+                validation_key = key
+
+        if not (train_key and test_key):
+            raise ValueError("Must have train and test inputs")
+
+        dataset_vars = {}
+
+        for each_key in [train_key, test_key, validation_key]:
+            if each_key:
+                dataset = kwargs[each_key]
+                if issubclass(self.python_interface.inputs[each_key], str):
+                    if "csv" in dataset:
+                        dataset_vars[each_key] = xgboost.DMatrix(
+                            dataset + "?format=csv&label_column=" + str(self._label_column)
+                        )
+                    else:
+                        dataset_vars[each_key] = xgboost.DMatrix(dataset)
+                elif issubclass(self.python_interface.inputs[each_key], FlyteFile):
+                    if FlyteContext.current_context().file_access.is_remote(dataset):
```

Comment:
i dont think you should do this, you should simply 
```python
with open(filename) as f:
    ....
```

Reply:
Since XGBoost accepts CSV, I think we can directly send the CSV file path. The same check is implemented in the [FlyteFile code](https://github.com/flyteorg/flytekit/blob/17201ce9725558f7c88e1f07ec4c97bc0ea31b6b/flytekit/types/file/file.py#L342) as well.

Reply:
Even if it is implemented in Flytefile, we may changed it. Can we simply use - https://github.com/flyteorg/flytekit/blob/17201ce9725558f7c88e1f07ec4c97bc0ea31b6b/flytekit/types/file/file.py#L208

dataset.download()? and that will give the path

Reply:
I have updated the code. I'm using `with open(dataset):` instead of download() because download() doesn't work with locally available FlyteFiles.